OtripleS.Web.Api/Services/StudentService.cs

```diff
@@ -26,5 +26,22 @@ public async ValueTask<Student> DeleteStudentAsync(Guid studentId)
 
             return await this.storageBroker.DeleteStudentAsync(maybeStudent);
         }
+
+        public async ValueTask<Student> RegisterAsync(Student student)
+        {
+            ValidateStudent(student);
+
+            try
+            {
+                student = await storageBroker.AddStudentAsync(student)
+                    .ConfigureAwait(false);
+                return student;
+            }
+            catch (Exception exception)
+            {
+                this.loggingBroker.LogError(exception);
+                throw;
+            }
+        }
```

Comment:
Exceptions should be handled in the exceptions partial class so the focus is mainly on the business logic not the exceptions - I did a bit of explanation around that point here:
https://www.youtube.com/watch?v=ECw-eYEG2HE

Reply:
updated as per video and merged changes in the RetrieveStudentByIdAsync method